src/libraries/System.Private.CoreLib/src/System/Reflection/InvokeUtils.cs

```diff
@@ -16,11 +16,16 @@ public static object ConvertOrWiden(Type srcType, CorElementType srcElementType,
 
             if (dstType.IsPointer)
             {
-                if (TryConvertPointer(srcObject, out IntPtr dstIntPtr))
+                if (TryConvertIntPtr(srcObject, out IntPtr dstIntPtr))
                 {
                     return dstIntPtr;
                 }
 
+                if (TryConvertUIntPtr(srcObject, out UIntPtr dstUIntPtr))
```

Comment:
Do we really need two methods? I think it would look better to just add `if (srcObject is UIntPtr srcUIntPtr)` case to the existing `TryConvertPointer` method.

Reply:
The extra boxing in this code is ok?

Reply:
If you mean about the `IntPtr=>object` the calling method was also boxing resulting `IntPtr` to object

Reply:
Yup. ok then :)